src/modules/search/hooks/useSearch.js

```diff
@@ -4,6 +4,10 @@ import { useTransactions } from '@transaction/hooks/queries';
 import { useValidators } from '@pos/validator/hooks/queries';
 import { useBlocks } from '@block/hooks/queries/useBlocks';
 
+function getIsLoading(queryRes) {
+  return queryRes.isLoading && queryRes.isFetching;
```

Comment:
isn't this boolean logic be an OR i.e 

```suggestion
  return queryRes.isLoading || queryRes.isFetching;
```

since if isLoading is true or isFetching is true meaning the entire query is still in a pending state

Reply:
What we want to do it to bring over the `isInitialLoading` logic which was introduced in v4.2 (we are still on v4.0) which keeps the isLoading state for spinners which were apparent in v3. 

This comment from the RQ author:
https://github.com/TanStack/query/issues/3584#issuecomment-1262638656

And this is how he implemented `isInitialLoading` 
https://github.com/TanStack/query/pull/4244/files#diff-144cf89d34f29ea1e30da14ec1cd6c323b645d0d446435aa1373b7d0fa00ccbc